### PR TITLE
feat(button): Allow Buttons to be built on custom components

### PR DESCRIPTION
**Pull request by ryardley**
Provide a "component" property to Button in order to pass in a custom html component name or a class reference.

### CODE REVIEW

**Comment by markdalgleish** [comment]
👏  This is fantastic.

Given how critical this functionality is, it's probably worth adding a test that ensures extraneous props are carried through, e.g. that you can render an `href` when using `component="a"`.

File: react/Button/Button.less [hunk 1]
```diff
@@ -17,6 +17,8 @@
 
 .root {
   .touchableText();
+  display: block;
```
Comment (markdalgleish) [inline react/Button/Button.less]: IIRC buttons are `inline-block` by default. I'm guessing we should probably change this?

**Comment by ryardley** [comment]
In testing I noticed that text was being highlighted on anchor buttons when being clicked. So added `user-select: none;` to Button.

**Comment by markdalgleish** [comment]
Awesome work, this was a much needed addition to the style guide 👍 

🎨 📦 🚀

**Comment by ryardley** [comment]
Thanks! :)

**Pull request merged**